plugin/src/main/java/org/opensearch/ml/action/batch/TransportBatchIngestionAction.java

```diff
@@ -131,33 +144,44 @@ protected void createMLTaskandExecute(MLBatchIngestionInput mlBatchIngestionInpu
             .state(MLTaskState.CREATED)
             .build();
 
-        mlTaskManager.createMLTask(mlTask, ActionListener.wrap(response -> {
-            String taskId = response.getId();
-            try {
-                mlTask.setTaskId(taskId);
-                mlTaskManager.add(mlTask);
-                listener.onResponse(new MLBatchIngestionResponse(taskId, MLTaskType.BATCH_INGEST, MLTaskState.CREATED.name()));
-                String ingestType = (String) mlBatchIngestionInput.getDataSources().get(TYPE);
-                Ingestable ingestable = MLEngineClassLoader.initInstance(ingestType.toLowerCase(), client, Client.class);
-                threadPool.executor(INGEST_THREAD_POOL).execute(() -> {
-                    executeWithErrorHandling(() -> {
-                        double successRate = ingestable.ingest(mlBatchIngestionInput);
-                        handleSuccessRate(successRate, taskId);
-                    }, taskId);
-                });
-            } catch (Exception ex) {
-                log.error("Failed in batch ingestion", ex);
-                mlTaskManager
-                    .updateMLTask(
-                        taskId,
-                        Map.of(STATE_FIELD, FAILED, ERROR_FIELD, MLExceptionUtils.getRootCauseMessage(ex)),
-                        TASK_SEMAPHORE_TIMEOUT,
-                        true
-                    );
-                listener.onFailure(ex);
+        mlModelManager.checkMaxBatchJobTask(mlTask, ActionListener.wrap(exceedLimits -> {
+            if (exceedLimits) {
+                String error = "exceed maximum BATCH_INGEST Task limits";
```

Comment:
Yes the message can be updated here. We can provide a link to hint CX to update the limits through settings, but it's better not to show the limits since some CX may consider it as customer specific and not public. This change is required the security team so we'd better be more cautious.